frontend-react/src/hooks/network/ManagePublicKey/ManagePublicKeyHooks.ts

```diff
@@ -0,0 +1,183 @@
+import { useReducer } from "react";
```

Comment:
Hmmm I don't think we need to follow the pattern in UseFileHandler.ts, and I'd argue that the logic there is a bit more complicated than it needs to be.  We're essentially working in a state machine in moving between the four "steps":
- Choose Sender
- Choose file
- Loading
- Result
    - Success (END)
    - Error (can reset to 'Choose file' step)

Each of these steps can be defined by inferred state:
- No Sender chosen --> render 'Choose Sender' step
- No file chosen + not yet submitted --> render 'Choose file' step
- File submitted and pending response --> render 'Loading' step
- Result
    - File submitted + successful response --> render 'Success' step
    - File submitted + errored response --> render 'Error' step

Rather than dispatching individual actions, we can derive this data from the actual form state + query state itself.

Reply:
I would greatly advise not using UseFileHandler as a template for derived work. I feel that hook is in need of review in the future.

Reply:
Refactored per comment